### PR TITLE
Move PHAR build to makefile and use newer version of box

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
Moving to the newer version of box with a view to scoping the dependencies in future

It struck me this isn't adequately tested so I've executed some of the behat tests against the PHAR - we could add more via a `@phar` tag

### CODE REVIEW

**Comment by jakzal** [comment]
Summoning @theofidry

**Comment by ciaranmcnulty** [comment]
@jakzal Not sure how to test I didn't break the deploy step in travis

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,3 @@
 behat.yml
 vendor
 composer.lock
-phpspec.phar
```
Comment (theofidry) [inline .gitignore:6]: shouldn't it remain ignored?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+box.phar:
```
Comment (theofidry) [inline Makefile:1]: you probably want to add `touch $@` at the end of each to ensure the file timestamp is properly updated

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	curl -Lls https://github.com/humbug/box/releases/download/3.8.4/box.phar -o box.phar
```
Comment (theofidry) [inline Makefile:2]: is it well understood here that if you change the link it is not enough to invalidate the existing `box.phar`?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	chmod +x box.phar
+
+phpspec.phar: box.phar vendor
```
Comment (theofidry) [inline Makefile:5]: I think it should also depend on the project source code, e.g. `$(shell find bin/ src/ -type f)` (see https://github.com/infection/infection/blob/master/Makefile#L208)

File: box.json [hunk 5]
```diff
@@ -1,5 +1,4 @@
 {
-    "chmod": "0755",
     "directories": [
```
Comment (theofidry) [inline box.json:2]: might be worth checking but you can move the current `files` to `files-bin` and remove completely `directories` and `finder`

File: .gitignore (hunk 1 above)
Comment (ciaranmcnulty) [reply .gitignore:6]: There's *.phar earlier in the file

File: box.json (hunk 5 above)
Comment (ciaranmcnulty) [reply box.json:2]: Thanks, I'll try it out

**Comment by ciaranmcnulty** [comment]
Thanks for the comments @theofidry - I'm not sure why invalidation is important in this case, as we're always making from a clean travis. Are you thinking about the developer who's made it locally and isn't doing `make clean` each time?

**Comment by ciaranmcnulty** [comment]
@theofidry And thanks for that hard work on box etc ;-)

**Comment by theofidry** [comment]
>Are you thinking about the developer who's made it locally and isn't doing make clean each time?

Yes: you'll likely have to create one and check it one locally at some point so having to clean it manually between each change might unnecessarily annoying

You're welcome :)

**Pull request merged**